common/lib/transport/connectionmanager.js

```diff
@@ -876,6 +871,15 @@ var ConnectionManager = (function() {
 		this.notifyState({state: 'closed'});
 	};
 
+	ConnectionManager.prototype.onAuthUpdated = function() {
+		var state = this.state.state;
+		if(state == 'connected' || state == 'connecting') {
+			/* in the current protocol version we are not able to update auth params on the fly;
+			 * so disconnect, and the new auth params will be used for subsequent reconnection */
+			this.disconnectAllTransports();
```

Comment:
> You might have a websocket connected over cellular, but by the time the reauth happens you're in the office on the office Wifi, where ws fails because of a proxy.

Sure, but why not try the same transport again and then fallback? I am reasonably confident that in most cases the transport that was used last could be used again, whereas assuming it can't in most cases negatively impacts performance, adds more work on both ends, and is unlikely to happen that often.